product-scenarios/1-manage-public-partner-private-apis/1.4-delete-an-existing-api/1.4.1-delete-an-existing-api/src/test/java/org/wso2/am/scenario/tests/delete/existing/api/DeleteAPINegativeTestCases.java

```diff
@@ -0,0 +1,223 @@
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case RETIRED:
+                updateRequest = new org.wso2.am.integration.test.utils.bean.APILifeCycleStateRequest(apiName,
+                        API_CREATOR_PUBLISHER_USER, APILifeCycleState.RETIRED);
+                verifyApiStatusChange(apiPublisher.changeAPILifeCycleStatus(updateRequest), state.toString());
+                break;
+            case CREATED:
+            default:
+//                    do nothing
+        }
+    }
+
+    private void subscribeToAPI(String apiName, String applicationName) throws Exception {
+        SubscriptionRequest subscriptionRequest = new SubscriptionRequest(apiName, API_CREATOR_PUBLISHER_USER);
+        subscriptionRequest.setApplicationName(applicationName);
+        subscriptionRequest.setTier(APIMIntegrationConstants.APPLICATION_TIER.UNLIMITED);
+        HttpResponse subscribeAPIResponse = apiStore.subscribe(subscriptionRequest);
+        verifyResponse(subscribeAPIResponse);
+    }
+
+    private void verifyApiStatusChange(HttpResponse apiUpdateResponse, String status) {
+        log.info("API life cycle state change response code : " + apiUpdateResponse.getResponseCode());
+        log.info("API life cycle state change response data : " + apiUpdateResponse.getData());
+        JSONArray updateStatus = new JSONObject(apiUpdateResponse.getData()).getJSONArray("lcs");
+        assertTrue(updateStatus.get(updateStatus.length() - 1).toString().contains("\"newStatus\":\"" + status + "\""),
+                "API life cycle state change failed");
+    }
+
+    private void checkDeleteAPI(String apiName, String errorMessage) throws Exception {
+        HttpResponse response = apiPublisher.deleteAPI(apiName, API_VERSION, API_CREATOR_PUBLISHER_USER);
+        log.info("API delete response code for API \'" + apiName + "\' : "
+                + response.getResponseCode());
+        log.info("API delete response data for API \'" + apiName + "\' : " + response.getData());
+        JSONObject responseData = new JSONObject(response.getData());
+        assertTrue(responseData.getBoolean("error"), "API deletion successful for API : " + apiName);
+        assertTrue(responseData.getString("message").contains(errorMessage),
+                "API deletion successful for API : " + apiName);
+    }
+
+    private void verifyAPIAvailableInPublisher(String apiName) throws Exception {
+        HttpResponse response = apiPublisher.getAPI(apiName, API_CREATOR_PUBLISHER_USER, API_VERSION);
+        log.info("Check API available in publisher response code for API \'" + apiName + "\' : "
+                + response.getResponseCode());
+        log.info("Check API available in publisher response data for API \'" + apiName + "\' : "
+                + response.getData());
+        assertTrue(response.getData().contains(apiName),
```

Comment:
If the response data contains apiName, does it mean that api is deleted?

Reply:
no, it means the API is still available in the publisher.

Reply:
ok. I got confused. since this is a negative test case, shall we add more context to error message so that it gives a better sense i.e. API has been deleted ...

Reply:
fixed